### ISSUE TITLE
Kucoin - Memory leak when running on development

### ISSUE THREAD

**Issue by rxlxrxsx** [CoinAlpha/hummingbot#3223]
**Describe the bug**
When running the Kucoin connector on the latest development, there's a significant increase in memory usage in a short period of time. So far I was able to run a pure_mm and avellaneda_mm but got the same result with memory usage. I'm also seeing these errors after stopping the strategy, similar to this [bug](https://github.com/CoinAlpha/hummingbot/issues/2627)

<img width="1424" alt="Screen Shot 2021-04-18 at 1 35 00 PM" src="https://user-images.githubusercontent.com/64377055/115135738-66a5cd00-a04d-11eb-8320-99325e07b40c.png">

Full traceback error can be seen on the attached log file below.

**Steps To Reproduce**

1. Run a pure_mm or avellaneda_mm using Kucoin as the connector.
2. After running, check for the memory usage using `htop` on your terminal.
3. Notice the increase in memory usage in a short period of time (CPU%, Mem and RES).
4. Cross checking with other connectors, I was only able to observe this behavior on Kucoin.

**Screenshots**
Attached screenshot and clips are shortly after running the strategy (pure_mm), for the samples on the master branch, I only attached an image since memory was stable upon running for quite some time.

Installation: Docker
Branch: Master
<img width="1360" alt="kucoin_docker_master_pure_mm" src="https://user-images.githubusercontent.com/64377055/115135985-4545e080-a04f-11eb-9a2e-cc942e3dcec0.png">

Installation: Docker
Branch: Development

https://user-images.githubusercontent.com/64377055/115136001-5bec3780-a04f-11eb-8dd6-5d2fe83a41e6.mp4

Installation: Source
Branch: Master
<img width="1423" alt="kucoin_pure_mm_master_source" src="https://user-images.githubusercontent.com/64377055/115136023-8211d780-a04f-11eb-946f-8d35fd237707.png">

Installation: Source
Branch: Development

https://user-images.githubusercontent.com/64377055/115136032-92c24d80-a04f-11eb-925d-40bd3ad4090c.mp4

**Release version**
```
branch: development
commit: 9e87685d54e344d4ee372db5e0baefa1644139f9
date: Fri Apr 16 16:43:59 2021 +0800
```

**Attachments**
Logs: [logs_kucoin_eth-usdt.log](https://github.com/CoinAlpha/hummingbot/files/6330576/logs_kucoin_eth-usdt.log)
Config: [kucoin_eth-usdt.txt](https://github.com/CoinAlpha/hummingbot/files/6330577/kucoin_eth-usdt.txt)

**Comment by rxlxrxsx** [comment]
After running the bot for 11.5 hrs, the instance became unresponsive. The only workaround is to manually kill the instance.

<img width="1423" alt="Screen Shot 2021-04-20 at 6 40 47 AM" src="https://user-images.githubusercontent.com/64377055/115336965-bd88df00-a1d2-11eb-9edd-aa34ef579fa4.png">

<img width="620" alt="Screen Shot 2021-04-20 at 6 39 22 AM" src="https://user-images.githubusercontent.com/64377055/115336987-c7aadd80-a1d2-11eb-96ce-eaa7b64b9b5a.png">

**Comment by dennisocana** [comment]
Thanks @rxlxrxsx for reporting. This is a defect in the current development branch that we'll prioritize fixing before the upcoming 0.39 release.

**Comment by RC-13** [comment]
fix on development branch(v0.39.0)
